### ISSUE TITLE
Wrong object keys in a spread operator

### ISSUE THREAD

**Issue by zamotkin** [javascript-obfuscator/javascript-obfuscator#494]
Options Transform Object Keys and Unicode Escape Sequence work wrong together in a spread operator.

## Minimal working example that will help to reproduce issue

```
const options = {
    ...NO_ADDITIONAL_NODES_PRESET,
    transformObjectKeys: true,
    unicodeEscapeSequence: true
}
```

### Before obfuscating
```
const {a} = {a: 1};
console.log(a);
```

### After obfuscating
```
const _0x43627d = {};
_0x43627d['\x5c\x78\x36\x31'] = 0x1;
const {a} = _0x43627d;
console['\x6c\x6f\x67'](a);
```

NB: I'm working on a fix.

**Comment by sanex3339** [comment]
Closed by https://github.com/javascript-obfuscator/javascript-obfuscator/pull/495

**Comment by sanex3339** [comment]
Released as `0.23.0`
